build_tools/docker/rbe_toolchain/Dockerfile

```diff
@@ -25,3 +25,18 @@ RUN apt-get update
 RUN apt-get install -y python3 python3-pip
 RUN python3 -m pip install --upgrade pip
 RUN python3 -m pip install numpy
+
+# Install dependencies for python3.6-dev
```

Comment:
Can we have an issue link for making the python upgrade here not gross? :-D

Reply:
Well, there's really only one other option that I can think of, which is to add the PPA manually. Hypothetically this should work
```dockerfile
RUN sed -i 's/python3/python3.5/g' /usr/bin/apt-add-repository
```
but it just... doesn't? It doesn't seem to have any discernable effect on what happens, even though `cat`ing out the file shows the header is definitely changed to `#! /usr/bin/python3.5`. `python3.5` is the right version (because our ugly solution works), so that's not the problem.

I can't seem to get adding the PPA manually to work however:
```dockerfile
RUN echo "deb http://ppa.launchpad.net/deadsnakes/ppa/ubuntu bionic main" >> /etc/apt/source.list
RUN echo "deb-src http://ppa.launchpad.net/deadsnakes/ppa/ubuntu bionic main" >> /etc/apt/source.list
RUN apt-key adv --keyserver keyserver.ubuntu.com --recv-keys F23C5A6CF475977595C89F51BA6932366A755776
RUN apt-get update
RUN apt-get install -y python3.6 python3.6-dev
# Reading package lists...
# Building dependency tree...
# Reading state information...
# E: Unable to locate package python3.6
# E: Couldn't find any package by glob 'python3.6'
# E: Couldn't find any package by regex 'python3.6'
# E: Unable to locate package python3.6-dev
# E: Couldn't find any package by glob 'python3.6-dev'
# E: Couldn't find any package by regex 'python3.6-dev'
# The command '/bin/sh -c apt-get install -y python3.6 python3.6-dev' returned a non-zero code: 100
```

Reply:
If this is based on some issue in the RBE container we're inheriting from, then we can tackle it there.